src/Uno.UI.Runtime.Skia.Gtk/Extensions/Storage/Pickers/FileSavePickerExtension.cs

```diff
@@ -0,0 +1,51 @@
+﻿#nullable enable
+
+using Gtk;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.UI.Runtime.Skia;
+using Windows.Storage;
+using Windows.Storage.Pickers;
+
+namespace Uno.Extensions.Storage.Pickers
+{
+	internal class FileSavePickerExtension : IFileSavePickerExtension
+	{
+		private readonly FileSavePicker _picker;
+
+		public FileSavePickerExtension(FileSavePicker owner)
+		{
+			_picker = owner ?? throw new ArgumentNullException(nameof(owner));
+		}
+
+		public async Task<StorageFile?> PickSaveFileAsync(CancellationToken token)
+		{
+			string commitText = "Save File";
+			if (!string.IsNullOrWhiteSpace(_picker.CommitButtonText))
+			{
+				commitText = _picker.CommitButtonText;
+			}
+
+			FileChooserDialog dialog = new FileChooserDialog(
+			"Save File",
+			GtkHost.Window,
+			FileChooserAction.Save,
+			"Cancel", ResponseType.Cancel,
+			commitText, ResponseType.Accept);
+
+			dialog.SelectMultiple = false;
+			dialog.SetFilename(_picker.SuggestedFileName);
+			dialog.SetCurrentFolder(PickerHelpers.GetInitialDirectory(_picker.SuggestedStartLocation));
```

Comment:
I'm not checking `FileTypeChoices` here, which most likely is a bug. Going to fix along with running/writing unit tests. Any hints for how to test? Thanks!